cosmos/providers/dbt/core/operators.py

```diff
@@ -215,75 +211,75 @@ def add_global_flags(self) -> list[str]:
+        if not source_path.exists():
+            raise AirflowException(f"Can not find the project_dir: {str(source_path)}")
+        if not source_path.is_dir():
+            raise AirflowException(
+                f"The project_dir {self.project_dir} must be a directory"
+            )
+        target_path = self.tmp_path.joinpath(os.path.basename(source_path))
+        if not target_path.exists():
+            target_path.mkdir(parents=True, exist_ok=True)
+        lock = FileLock(target_path / ".lock", timeout=10)
+        with suppress(Timeout):
+            lock.acquire(timeout=15)
             comparison = dircmp(
-                self.project_dir, target_dir, ignore=["logs", "target", ".lock"]
-            )  # compares tmp and project dir
-            changes = has_differences(comparison)  # check for changes
-
-        # if there are changes between tmp and project_dir then copy them over
-        if changes:
-            logging.info(
-                f"Changes detected - copying {self.project_dir} to {target_dir}"
+                source_path, target_path, ignore=["logs", "target", ".lock"]
             )
-            # put a lock on the dest so that it isn't getting written multiple times
-            lock_file = os.path.join(target_dir, ".lock")
-
-            # if there is already a lock file - then just wait for it to be released and continue without copying
-            if os.path.exists(lock_file) and is_file_locked(lock_file):
-                while os.path.exists(lock_file):
-                    with open(lock_file, "w") as lock_f:
-                        try:
-                            # Lock acquired, the lock file is available
-                            fcntl.flock(lock_f, fcntl.LOCK_SH | fcntl.LOCK_NB)
-                            break
-                        except OSError:
-                            # Lock is held by another process, wait and try again
-                            time.sleep(1)
-
-                # The lock file is available, release the shared lock
-                with open(lock_file, "w") as lock_f:
-                    fcntl.flock(lock_f, fcntl.LOCK_UN)
-
-            # otherwise create a lock file and copy the dbt directory
+            if has_differences(comparison):
+                logging.info(
+                    f"Changes detected - copying {str(source_path)} to {str(target_path)}"
+                )
+                for dir_object in os.listdir(target_path):
+                    dir_object: str
+                    child_path = target_path.joinpath(dir_object)
+                    if child_path.is_dir():
+                        if dir_object not in ["logs", "target"]:
+                            shutil.rmtree(child_path)
+                    elif dir_object not in [".lock"]:
+                        os.remove(child_path)
+                shutil.copytree(
+                    source_path, target_path, ignore=exclude, dirs_exist_ok=True
```

Comment:
hey just wanted to comment here that the `dirs_exist_ok` arg is not available in python 3.7, so if it's required the minimum python version for cosmos in the pyproject [here](https://github.com/astronomer/astronomer-cosmos/blob/b8fbe4c463725e7bbc0ef29b4bfecfb6c0295b2f/pyproject.toml#L11) should be 3.8

Reply:
Thanks @jbandoro, I've adjusted the test environment to be python 3.7 so it supports the minimum version so hopefully I don't make that same mistake again!
Just about to push up a commit that makes it 3.7 compliant.